cognite/neat/core/extractors/rdf_to_assets.py

```diff
@@ -838,17 +845,27 @@ def _micro_batch_push(
     """
     total = len(assets)
     counter = 0
-
+    if push_type not in ["update", "create"]:
+        logging.info(f"push_type {push_type} not supported")
+        raise ValueError(f"push_type {push_type} not supported")
     for batch in chunker(assets, batch_size):
         counter += len(batch)
         start_time = datetime_utc_now()
-        if push_type == "update":
-            client.assets.update(batch)
-        elif push_type == "create":
-            client.assets.create_hierarchy(batch)
-        else:
-            logging.info(f"push_type {push_type} not supported")
-            raise ValueError(f"push_type {push_type} not supported")
+        retry_counter = 0
+        while True:
+            try:
+                if push_type == "update":
+                    client.assets.update(batch)
+                elif push_type == "create":
+                    client.assets.create_hierarchy(batch)
+                break
+            except Exception as e:
+                if retry_counter >= max_retries:
+                    raise e
+                retry_counter += 1
+                logging.error(f"Error pushing {push_type} micro-batch: {e}")
+                logging.info(f"Retrying ({retry_counter})  in {retry_delay} seconds")
+                time.sleep(retry_delay)
```

Comment:
Worth extracting this logic into a decorator?

Reply:
fixed